internal/lint/publish.go

```diff
@@ -44,17 +45,46 @@ func Publisher(input *PublisherInput) {
 	})
 
 	ginkgo.It("should publish", func() {
-		Ω.Expect(subject.Topic(topicA).Publish(ctx, &bps.PubMessage{Data: []byte("v1")})).To(Ω.Succeed())
-		Ω.Expect(subject.Topic(topicB).Publish(ctx, &bps.PubMessage{Data: []byte("v2")})).To(Ω.Succeed())
-		Ω.Expect(subject.Topic(topicA).Publish(ctx, &bps.PubMessage{Data: []byte("v3")})).To(Ω.Succeed())
+		// run publishing and consuming concurrently (mostly for StartPosition=PositionNewest case):
+		var threads sync.WaitGroup
 
-		Ω.Eventually(func() ([]*bps.PubMessage, error) {
-			return input.Messages(topicA, 2)
-		}, "3s").Should(haveData("v1", "v3"))
+		// produce concurrently:
+		threads.Add(1)
+		go func() {
+			defer threads.Done()
+			defer ginkgo.GinkgoRecover()
```

Comment:
nit: that should come first, before done, it;s the last thing we want to do

Reply:
I'd prefer `panic` not to break `threads.Done()`, but ok